protomongo/protomongo.go

```diff
@@ -139,8 +141,12 @@ type protoHelper struct {
 func (pc *protobufCodec) protoHelper(pb descriptor.Message, t reflect.Type) *protoHelper {
 	// Try to load a pre-existing protoHelper from cache, if it exists.
 	messageName := proto.MessageName(pb)
-	if ph, ok := pc.protoHelpers[messageName]; ok {
-		return ph
+
+	pc.mux.Lock()
```

Comment:
don't bother locking/unlocking on read; the vast majority of the time the read will return successfully so this will just cost time.